pkg/pillar/zfs/zfs.go

```diff
@@ -78,6 +91,62 @@ func CreateVaultDataset(datasetName, zfsKeyFile string) error {
 	return MountDataset(datasetName)
 }
 
+// GetDatasetAvailableBytes Read Zfs dataset 'available' space property, parse as uint64
+func GetDatasetAvailableBytes(datasetName string) (uint64, error) {
+	ds, err := libzfs.DatasetOpen(datasetName)
+	if err != nil {
+		return 0, fmt.Errorf("Open dataset %s failure: %v", datasetName, err)
+	}
+	defer ds.Close()
+	zpoolPropAvailBytes, err := ds.GetProperty(libzfs.DatasetPropAvailable)
+	if err != nil {
+		return 0, fmt.Errorf("Read dataset %s available space failure: %v", datasetName, err)
+	}
+	size, err := strconv.ParseUint(zpoolPropAvailBytes.Value, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("dataset %s available space parse failure: %v", datasetName, err)
+	}
+	return size, nil
+}
+
+// GetZvolPath Helper to build the /dev/zvol/<dataset> path
+func GetZvolPath(datasetName string) string {
+	return types.ZVolDevicePrefix + "/" + datasetName
```

Comment:
If you care about not getting duplicated "//" in the path name then better to use the filepath package to do the concatenation.